File: game_server/ServerProtocol/src/GS2GC/p014_ChildOp/GS2GC_014_012_RetGetRecommendPlayerList.java
package GS2GC.p014_ChildOp;

import java.nio.ByteBuffer;
/*********
 * 联姻请求-获取推荐玩家列表
 **/
public class GS2GC_014_012_RetGetRecommendPlayerList implements ALBasicProtocolPack._IALProtocolStructure {
/** 空 */
private java.util.ArrayList<Common.ChildObj.Adult_PoolBaseInfo> matchList;


public GS2GC_014_012_RetGetRecommendPlayerList() {
	matchList = new java.util.ArrayList<Common.ChildObj.Adult_PoolBaseInfo>();
}

public GS2GC_014_012_RetGetRecommendPlayerList(
	 java.util.ArrayList<Common.ChildObj.Adult_PoolBaseInfo> _matchList
) {	matchList = _matchList;
}

public final byte getMainOrder() { return (byte)14; }

public final byte getSubOrder() { return (byte)12; }

/** 空 */
public java.util.ArrayList<Common.ChildObj.Adult_PoolBaseInfo> getMatchList() { return matchList; }
/** 空 */
public void addMatchList(Common.ChildObj.Adult_PoolBaseInfo _matchList) { matchList.add(_matchList); }


public final int GetBufSize() {
	int _size = 0;
	_size += 2 + (matchList.size() * 36);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 2;
	_size += 2 + (matchList.size() * 36);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _matchListCount = _buf.getShort();
	for(int _i = 0; _i < _matchListCount; _i++) { 
		Common.ChildObj.Adult_PoolBaseInfo _matchList = new Common.ChildObj.Adult_PoolBaseInfo();
		if(_buf.remaining() <= 0) return;
	int __matchListCustLen = _buf.getInt();
	int __matchListCurPos = _buf.position();
	_matchList.ReadUnzipBuf(_buf, __matchListCurPos + __matchListCustLen);
	_buf.position(__matchListCurPos + __matchListCustLen);

		matchList.add(_matchList);
	}
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putShort((short)matchList.size());
	for(int _i = 0; _i < matchList.size(); _i++) { 
		_buf.putInt(matchList.get(_i).GetBufSize());
	matchList.get(_i).PutUnzipBuf(_buf);
	}
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)14);
	_buf.put((byte)12);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)14);
	_recBuf.put((byte)12);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

File: game_server/ClientProtocol/GS2GC/p014_ChildOp/GS2GC_014_012_RetGetRecommendPlayerList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GS2GC.p014_ChildOp
{

/// <summary>
/// 联姻请求-获取推荐玩家列表
/// </summary>
public class GS2GC_014_012_RetGetRecommendPlayerList : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 空
/// </summary>
private List<Common.ChildObj.Adult_PoolBaseInfo> matchList;


public GS2GC_014_012_RetGetRecommendPlayerList() {
	matchList = new List<Common.ChildObj.Adult_PoolBaseInfo>();
}

public GS2GC_014_012_RetGetRecommendPlayerList(
	List<Common.ChildObj.Adult_PoolBaseInfo> _matchList
) {	matchList = _matchList;
}

public byte getMainOrder() { return (byte)14; }

public byte getSubOrder() { return (byte)12; }

/// <summary>
/// 空
/// </summary>
public List<Common.ChildObj.Adult_PoolBaseInfo> getMatchList() { return matchList; }
/// <summary>
/// 空
/// </summary>
public void addMatchList(Common.ChildObj.Adult_PoolBaseInfo _matchList) { matchList.Add(_matchList); }


public int GetBufSize() {
	int _size = 0;
	_size += 2 + (matchList.Count * 36);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 2;
	_size += 2 + (matchList.Count * 36);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _matchListCount = _buf.getShort();
	for(int _i = 0; _i < _matchListCount; _i++) { 
		Common.ChildObj.Adult_PoolBaseInfo _matchList = new Common.ChildObj.Adult_PoolBaseInfo();
		int __matchListCustLen = _buf.getInt();
	int __matchListCurPos = _buf.getCurPos();
	_matchList.ReadUnzipBuf(_buf, __matchListCurPos + __matchListCustLen);
	_buf.setPosition(__matchListCurPos + __matchListCustLen);

		matchList.Add(_matchList);
	}
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putShort((short)matchList.Count);
	for(int _i = 0; _i < matchList.Count; _i++) { 
		_buf.putInt(matchList[_i].GetBufSize());
	matchList[_i].PutUnzipBuf(_buf);
	}
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)14);
	_buf.put((byte)12);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)14);
	_recBuf.put((byte)12);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("matchList").Append(":").Append(matchList.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

